### DISCUSSION TITLE
Different paths for files you should edit vs generated files

### DISCUSSION THREAD

**Discussion by osdiab** [ivanhofer/typesafe-i18n#719] [DX improvements]
Right now you set a single `outputPath` like `src/i18n` and everything is placed there.

But the actual translations files, and the formatters, are things you should be editing, whereas the other stuff seems to be things that you mostly just auto-generate. This would be easier to understand as a team if the files you touch can be located in `src/i18n` and the files that are generated get placed at e.g. `gen/i18n` so that they're not right next to each other.

for now will just add a `gen-` prefix to the generated stuff to make it clear.

**Comment by ivanhofer** [comment]
You can already manually override the names of each file: https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/generator#options. You can set the `typesFileName` and `utilFileName` to include the `gen-` prefix.

**Reply by osdiab** [reply]
I’m saying to put them in totally different directories, I find it confusing to comingle generated files vs files you’re supposed to edit. So not just some prefix but a straight up different output directory, id rather my src directory has no generated files whatsoever. Not a big deal but yeah

**Reply by ivanhofer** [reply]
You should be able to do that already with the `*FileName` options.
Making this a default would require a breaking change and I don't see the benefit as you are the first one complaining.
Maybe we can add this to the next major release.

But I'm open to a PR that makes it possible to have better support for renaming the adapter files like you wrote on discord.
